File: TextosJapones/src/main/java/org/jisho/textosJapones/model/entities/Kanji.java
package org.jisho.textosJapones.model.entities;

import java.util.UUID;

public class Kanji {
    private UUID id;
    private String kanji;
    private String palavra;
    private String significado;

    public UUID getId() {
        return id;
    }

    public void setId(UUID id) {
        this.id = id;
    }

    public String getKanji() {
        return kanji;
    }

    public void setKanji(String kanji) {
        this.kanji = kanji;
    }

    public String getPalavra() {
        return palavra;
    }

    public void setPalavra(String palavra) {
        this.palavra = palavra;
    }

    public String getSignificado() {
        return significado;
    }

    public void setSignificado(String significado) {
        this.significado = significado;
    }

    public Kanji(UUID id, String kanji, String palavra, String significado) {
        this.id = id;
        this.kanji = kanji;
        this.palavra = palavra;
        this.significado = significado;
    }
}

File: ProcessaTexto/src/main/kotlin/br/com/fenix/processatexto/model/entities/processatexto/Kanji.kt
package br.com.fenix.processatexto.model.entities.processatexto

import br.com.fenix.processatexto.model.entities.EntityBase
import jakarta.persistence.*
import org.hibernate.annotations.JdbcTypeCode
import org.hibernate.type.SqlTypes
import java.util.*

@Table(name = "kanjax_pt")
data class Kanji(
    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    @JdbcTypeCode(SqlTypes.VARCHAR)
    @Column(name = "ID", nullable = false)
    private val id: UUID?,
    @Column(name = "kanji")
    val kanji: String,
    @Column(name = "palavra")
    val palavra: String,
    @Column(name = "significado")
    val significado: String
) : EntityBase<UUID?, Kanji>() {

    override fun getId(): UUID? = id
    override fun create(id: UUID?): Kanji = Kanji()

    constructor() : this(null, "", "", "")

}